hyperspy/misc/io/fei_stream_readers.py

```diff
@@ -193,24 +193,33 @@ def stream_to_sparse_COO_array(
         If True, sum all the frames
 
     """
-    if sum_frames:
-        coords, data, shape = _stream_to_sparse_COO_array_sum_frames(
-            stream_data=stream_data,
-            shape=spatial_shape,
-            channels=channels,
-            rebin_energy=rebin_energy,
-            first_frame=first_frame,
-            last_frame=last_frame,
-        )
+    # workaround for empty stream, numba "doesn't support" empty list, see
+    # https://github.com/numba/numba/pull/2184
+    if np.allclose(stream_data, np.full_like(stream_data, 65535)):
```

Comment:
Wouldn't it be possible to use a workaround such as the one in https://github.com/numba/numba/issues/2152#issuecomment-254520288 instead? This one is fine when the array is indeed empty, but when it is not (the most common and useful case) it requires going twice through the data.

Reply:
Indeed, this doubles the memory requirement when loading data... it should be fixed now.